crates/iroha_wasm_builder/src/main.rs

```diff
@@ -42,24 +42,56 @@ fn main() -> color_eyre::Result<()> {
             common: CommonArgs { path },
             profile,
         } => {
-            let builder = Builder::new(&path, &profile).show_output();
+            let profile: Profile = profile.parse().expect("Invalid profile provided");
+            let builder = Builder::new(&path, profile).show_output();
             builder.check()?;
         }
         Cli::Build {
             common: CommonArgs { path },
             out_file,
             profile,
         } => {
-            let builder = Builder::new(&path, &profile).show_output();
+            let profile: Profile = profile.parse().expect("Invalid profile provided");
+            let builder = Builder::new(&path, profile).show_output();
 
             let output = {
                 // not showing the spinner here, cargo does a progress bar for us
-                match builder.build() {
+                match builder.build_unoptimized() {
```

Comment:
I think in this block the output can be finalized to be optimized or not, by somehow branching the build method depending on stdin i.e. `main` is responsible for interpreting profiles and `lib` is for optimizing outputs

Reply:
Currently, `build` decides whether optimizations are needed and applies them. The problem `build_unoptimized` solving is the need to insert a spinner between build and optimization.